src/Pos/Block/Logic/Slog.hs

```diff
@@ -144,7 +143,7 @@ slogVerifyBlocks blocks = do
 -- | Set of constraints necessary to apply/rollback blocks in Slog.
 type SlogApplyMode ssc m =
     ( SlogMode ssc m
-    , MonadBlockDBWrite ssc m
+    , MonadBlockDBWrite (BlockHeader ssc) (Block ssc) Undo m
```

Comment:
It's inconvenient to write `MonadBlockDBWrite (BlockHeader ssc) (Block ssc) Undo` and it's actually the only reasonable set of type parameters. Can you define an alias?